From 7d4d0bfc9cad9897321bc9ec65f45794e4486cd1 Mon Sep 17 00:00:00 2001
From: Kaviarasu Sanjeevi <quic_kaviaras@quicinc.com>
Date: Tue, 17 Sep 2024 15:48:45 +0530
Subject: [PATCH] ath11k: Compilation warning fixes for ath11k Protocol files

reg.c    - Modified the datatype of max_elements to u32 from u8 and also in caller function
nss.c    - Modified the datatype of addr to const u8 * in ath11k_nss_peer_delete
nss.h    - Modified the datatype of addr to const u8 * in ath11k_nss_peer_delete

Signed-off-by: Kaviarasu Sanjeevi <quic_kaviaras@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/reg.c  | 3 ++-
 drivers/net/wireless/ath/ath11k/nss.c  | 2 +-
 drivers/net/wireless/ath/ath11k/nss.h  | 4 ++--
 3 files changed, 5 insertions(+), 4 deletions(-)

--- a/drivers/net/wireless/ath/ath11k/reg.c
+++ b/drivers/net/wireless/ath/ath11k/reg.c
@@ -442,7 +442,7 @@ ath11k_ieee80211_ap_pwr_type_convert(enu
 
 static struct cur_reg_rule
 *ath11k_get_active_6g_reg_rule(struct cur_regulatory_info *reg_info,
-			       u32 *max_bw_6g, int *max_elements,
+			       u32 *max_bw_6g, u32 *max_elements,
 			       enum nl80211_regulatory_power_modes *pwr_mode)
 {
 	struct cur_reg_rule *reg_rule = NULL;
@@ -484,7 +484,8 @@ ath11k_reg_build_regd(struct ath11k_base
 {
 	struct ieee80211_regdomain *new_regd = NULL;
 	struct cur_reg_rule *reg_rule, *reg_rule_6g;
-	u8 i = 0, j = 0, k = 0, max_elements = 0;
+	u8 i = 0, j = 0, k = 0;
+	u32 max_elements = 0;
 	u8 num_rules;
 	u16 max_bw;
 	u32 flags, reg_6g_number = 0, max_bw_6g = 0;
--- a/drivers/net/wireless/ath/ath11k/nss.c
+++ b/drivers/net/wireless/ath/ath11k/nss.c
@@ -3642,7 +3642,7 @@ void ath11k_nss_update_sta_rxrate(struct
 	peer->nss.nss_stats->rxrate.bw = ath11k_mac_bw_to_mac80211_bw(ppdu_info->bw);
 }
 
-int ath11k_nss_peer_delete(struct ath11k_base *ab, u32 vdev_id, u8 *addr)
+int ath11k_nss_peer_delete(struct ath11k_base *ab, u32 vdev_id, const u8 *addr)
 {
 	struct nss_wifili_peer_msg *peer_msg;
 	struct nss_wifili_msg *wlmsg = NULL;
--- a/drivers/net/wireless/ath/ath11k/nss.h
+++ b/drivers/net/wireless/ath/ath11k/nss.h
@@ -281,7 +281,7 @@ int ath11k_nss_vdev_create(struct ath11k
 void ath11k_nss_vdev_delete(struct ath11k_vif *arvif);
 int ath11k_nss_vdev_up(struct ath11k_vif *arvif);
 int ath11k_nss_vdev_down(struct ath11k_vif *arvif);
-int ath11k_nss_peer_delete(struct ath11k_base *ab, u32 vdev_id, u8 *addr);
+int ath11k_nss_peer_delete(struct ath11k_base *ab, u32 vdev_id, const u8 *addr);
 int ath11k_nss_set_peer_authorize(struct ath11k *ar, u16 peer_id);
 int ath11k_nss_peer_create(struct ath11k *ar, struct ath11k_peer *peer);
 void ath11k_nss_peer_stats_enable(struct ath11k *ar);
@@ -382,7 +382,7 @@ static inline int ath11k_nss_vdev_down(s
 }
 
 static inline int ath11k_nss_peer_delete(struct ath11k_base *ab, u32 vdev_id,
-					 u8 *addr)
+					 const u8 *addr)
 {
 	return 0;
 }
